actionview/lib/action_view/renderer/partial_renderer.rb

```diff
@@ -521,7 +521,7 @@ def retrieve_template_keys
     def retrieve_variable(path, as)
       variable = as || begin
         base = path[-1] == "/".freeze ? "".freeze : File.basename(path)
-        raise_invalid_identifier(path) unless base =~ /\A_?(.*)(?:\.\w+)*\z/
+        raise_invalid_identifier(path) unless base =~ /\A_?(.*?)(\.\w+)*\z/
```

Comment:
True. I added the `?:` to express that we're only using `$1` afterwards.
It would work without `?:`, or you can make another PR adding `?:`, or I would just push these 2 chars.
I'm fine with any option :)